components/ContentItem.tsx

```diff
@@ -0,0 +1,209 @@
+  descriptions: Description[]
+  version: string
+  citationCount: number
+  viewCount: number
+  downloadCount: number
+}
+
+interface Creator {
+  id: string
+  name: string
+  givenName: string
+  familyName: string
+}
+
+interface Title {
+  title: string
+}
+
+interface Description {
+  description: string
+}
+
+const ContentItem: React.FunctionComponent<Props> = ({item}) => {
+  const title = () => {
+    if (!item.titles[0]) return (
+      <h3 className="member">No Title</h3>
+    )
+
+    const titleHtml = item.titles[0].title 
+    
+    return (
+      <h3 className="member">
+        {ReactHtmlParser(titleHtml)}
+        {item.types.resourceTypeGeneral &&
+          <span className="small"> {startCase(item.types.resourceTypeGeneral)}</span>
+        }
+      </h3>
+    )
+  }
+
+  const creators = () => {
+    if (!item.creators) return 'No creators'
+
+    const creatorList = item.creators.reduce( (sum, creator, index, array) => {
+      const c = creator.familyName ? [creator.givenName, creator.familyName].join(' ') : creator.name
+      
+      // padding depending on position in creators list
+      if (array.length > index + 2) {
+        return sum + c + ', '
+      } else if (array.length > index + 1) {
+        return sum + c + ' & '
+      } else {
+        return sum + c
+      }
+    }, '')
+
+    return (
+      <div className="creators">
+        {creatorList}
+      </div>
+    )
+  }
+
+  const metadata = () => {
+    return (
+      <div className="metadata">
+        {item.version ? 'Version ' + item.version + ' of ' : ''}{item.types.resourceType ? startCase(item.types.resourceType) : 'Content'} published {item.publicationYear} via {item.publisher}
+      </div>
+    )
+  }
+
+  const description = () => {
+    if (!item.descriptions[0]) return ''
+
+    const descriptionHtml = truncate(item.descriptions[0].description, { 'length': 750, 'separator': '… '})
+
+    return (
+      <div className="description">
+        {ReactHtmlParser(descriptionHtml)}
+      </div>
+    )
+  }
+
+  const metricsCounter = () => {
+    if (item.citationCount + item.viewCount + item.downloadCount == 0) {
+      return (
+        <div className="metrics-counter">
+          <i><FontAwesomeIcon icon={faInfoCircle}/> No citations, views or downloads reported.</i>
+        </div>
+      )
+    }
+
+    return (
+      <div className="metrics-counter">
+        {item.citationCount > 0 &&
+          <i><FontAwesomeIcon icon={faQuoteLeft}/> <Pluralize singular={'Citation'} count={item.citationCount} /> </i>
+        }
+        {item.viewCount > 0 &&
+          <i><FontAwesomeIcon icon={faEye}/> <Pluralize singular={'View'} count={item.viewCount} /> </i>
+        }
+        {item.downloadCount > 0 &&
+          <i><FontAwesomeIcon icon={faDownload}/> <Pluralize singular={'Download'} count={item.downloadCount} /> </i>
+        }
+      </div>
+    )  
+  }
+
+  const bookmark = (
+    <Popover id="bookmark" title="Bookmarking">
```

Comment:
Is there a separated user story for this feature? I do not think its in the list. https://docs.google.com/spreadsheets/d/1EnnNmm1RwXUuDovCnfKViDcNJ_KItJr9l-yYlUutozs/edit#gid=0

Reply:
This needs discussion. And the user story list is far from complete.